### PR TITLE
Core: updated efsw to newest version

### PR DESCRIPTION
**Pull request by damaex**
**Changes proposed:**

-  updated efsw to newest version from https://bitbucket.org/SpartanJ/efsw

**Target branch(es):** 3.3.5/master

- [x] 3.3.5

**Issues addressed:** Closes #  (insert issue tracker number)

**Tests performed:** (Does it build, tested in-game, etc.)
- [x] build
    - [x] static
    - [x] dynamic
- [x] in game (on raspberry pi 4)

**Known issues and TODO list:** (add/remove lines as needed)

### CODE REVIEW

File: dep/efsw/CMakeLists.txt [hunk 1]
```diff
@@ -1,88 +1,112 @@
...
-    src/efsw/FileWatcherGeneric.cpp
-    src/efsw/FileWatcherImpl.cpp
-    src/efsw/Log.cpp
-    src/efsw/Mutex.cpp
-    src/efsw/String.cpp
-    src/efsw/System.cpp
-    src/efsw/Thread.cpp
-    src/efsw/Watcher.cpp
-    src/efsw/WatcherGeneric.cpp)
+          src/efsw/Debug.cpp
```
Comment (jackpoz) [inline dep/efsw/CMakeLists.txt]: are all these changes part of the upgraded library or are they TC-specific ones ?

Comment (Shauren) [reply dep/efsw/CMakeLists.txt]: Looks like upstream changes to me https://bitbucket.org/SpartanJ/efsw/src/default/CMakeLists.txt

File: dep/efsw/CMakeLists.txt [hunk 2]
```diff
@@ -1,88 +1,112 @@
...
+            "src/efsw/WatcherFSEvents.cpp"
+            "src/efsw/FileWatcherInotify.cpp"
+            "src/efsw/FileWatcherWin32.cpp"
+            "src/efsw/FileWatcherFSEvents.cpp")
+  endif ()
 
-      set(OPTIONAL_MAC_LINK_LIBRARIES "-framework CoreFoundation" "-framework CoreServices")
-    endif()
-  endif()
+  if (MSVC)
```
Comment (Shauren) [inline dep/efsw/CMakeLists.txt]: While it is part of original CMakeLists.txt on bitbucket, you need to remove this part - we don't want to have warnings (or any compile flags actually) messed with in /dep

**Comment by jackpoz** [comment]
Anything left to do before being able to merge this PR ?

**Comment by Shauren** [comment]
Needs to be tested that script reloading still works - PR only mentions that logging ingame worked

File: dep/efsw/CMakeLists.txt (hunk 1 above)
Comment (Aokromes) [reply dep/efsw/CMakeLists.txt]: it's overjustified compared to that repo.

**Comment by damaex** [comment]
Just tested the script reloading on Raspberry Pi 4 (4GB):
- started worldserver build with dynamic linkage
- added newline to the end of `/src/server/scripts/Pet/pet_priest.cpp`
- worldserver picks up the change and recompiles the `scripts_pet` library
- worldserver loads the library

```
[ 38%] Built target shared

[ 44%] Built target efsw

[ 97%] Built target game

Scanning dependencies of target scripts_pet

[ 97%] Building CXX object src/server/scripts/CMakeFiles/scripts_pet.dir/Pet/pet_priest.cpp.o

[ 97%] Linking CXX shared library libscripts_pet.so

[100%] Built target scripts_pet

>> Successfully build module Pet, continue with installing...
Starting asynchronous install job for module Pet...
-- Install configuration: "RelWithDebInfo"

-- Install component: "scripts_pet"

-- Installing: /home/ubuntu/server/bin/scripts/libscripts_pet.so

-- Set runtime path of "/home/ubuntu/server/bin/scripts/libscripts_pet.so" to "/home/ubuntu/server/lib"

>> Successfully installed module Pet in 13s
Released script module "libscripts_pet.so" ("Pet")...
>> Loaded script module "libscripts_pet.so" ("Pet" - b376a68).
>> Validated 2767 scripts in 8 ms
```

File: dep/efsw/CMakeLists.txt [hunk 3]
```diff
@@ -18,50 +18,48 @@ if (BUILD_SHARED_LIBS)
...
-  else()
-    list(APPEND SRCS
+  else ()
+    list (APPEND SRCS
       src/efsw/platform/posix/FileSystemImpl.cpp
       src/efsw/platform/posix/MutexImpl.cpp
       src/efsw/platform/posix/SystemImpl.cpp
       src/efsw/platform/posix/ThreadImpl.cpp)
-  endif()
+  endif()
```
Comment (Aokromes) [inline dep/efsw/CMakeLists.txt]: remove whitespace

**Comment by Shauren** [comment]
I hope I didn't break it for your raspberry needs with that CMakeLists.txt change

**Comment by damaex** [comment]
Nope, works well on raspberry with these changes.

**Pull request merged**